src/service.js

```diff
@@ -182,12 +183,23 @@ class Service {
       data[this.id] = id;
     }
 
-    return this.Model
-      .update(params.query, { $set: data }, options)
-      .lean(this.lean)
-      .exec()
-      .then(() => this._getOrFind(id, params))
-      .catch(errorHandler);
+    let promise;
+
+    // We need this shitty hack because update doesn't return
+    // a promise properly when runValidators is true. WTF!
+    try {
+      promise = this.Model
```

Comment:
Ya I think https://github.com/feathersjs/feathers-service-tests/blob/master/src/common-tests.js#L429-L436 covers the patch case. I don't think we need `$set`. I was concerned with the whole object being updated but [Mongoose actually does that internally for you](http://mongoosejs.com/docs/api.html#model_Model.update) when `overwrite: false` (which is the default).